plugins/websearch/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+use abi_stable::std_types::{ROption, RString, RVec};
+use anyrun_plugin::*;
+use serde::Deserialize;
+use std::{fs, process::Command};
+use urlencoding::encode;
+
+#[derive(Deserialize)]
+struct Config {
+    prefix: String,
+    engine: String,
```

Comment:
If there are a specific set of engines, like seen in the handler, you should use an enum for this.

Reply:
Hm, i'm not that familiar with Rust and i struggle to define an Enum as a value for RON.

I've changed it so you can define multiple search engines, so i'd wanna do something like `Vec<Engine>` as the config struct value, but this doesnt work. No error sadly.

Reply:
You can simply define the search engine as an enum like this:
```rs
#[derive(Deserialize)]
enum SearchEngine {
    Google,
    DuckDuckGo,
    Bing,
}```

Then you can place it in the config struct, `serde` will deal with the deserialization of it automatically.

Reply:
Hm, I see no problem with the code and neither does my compiler. Specifying multiple search engines seems a bit weird though.

Reply:
well, yeah...some people wanna use multiple search engines though. dont ask me why. they do.

in the end you can specify one if you wish.